CHANGELOG.md

```diff
@@ -1,3 +1,7 @@
+## 7.4.0
```

Comment:
can we break this into two prs with a single concern? makes it easier to ship 7.3.1 with the bug fix, and 7.4.0 with the fix + new shutdown strategy